File: C# examples/ThreadSpecificStoragePattern/Storage.cs
namespace ThreadSpecificStoragePattern
{
    public static class Storage
    {
        private static ThreadLocal<Data> _threadLocalData = new ThreadLocal<Data>(() => new Data());

        // not a thread-specific data
        // private static Data _threadLocalData = new Data();

        public static Data GetData()
        {
            return _threadLocalData.Value;
        }
    }
}

File: Java examples/src/threadSpecificStorage/Storage.java
package threadSpecificStorage;

public class Storage {
    private static ThreadLocal<Data> threadLocalData = ThreadLocal.withInitial(Data::new);
    // not a thread-specific data
    // private static Data threadLocalData =  new Data();

    public static Data getData() {
        return threadLocalData.get();
    }
}
